src/shapes/star.cpp

```diff
@@ -0,0 +1,44 @@
+#include "math/pi.hpp"
+#include "shapes/star.hpp"
+#include "shapes/straight_vertex.hpp"
+#include <cmath>
+
+using namespace rive;
+
+void Star::innerRadiusChanged() { markPathDirty(); }
+
+void Star::update(ComponentDirt value)
+{
+	if (hasDirt(value, ComponentDirt::Path))
+	{
+		clearVertices();
+
+		auto actualPoints = points() * 2;
+		auto halfWidth = width() / 2;
+		auto halfHeight = height() / 2;
+		auto innerHalfWidth = width() * innerRadius() / 2;
+		auto innerHalfHeight = height() * innerRadius() / 2;
+
+		auto angle = -pi / 2;
+		auto inc = 2 * pi / actualPoints;
+
+		for (int i = 0; i < points(); i++)
+		{
+			StraightVertex* outerVertex = new StraightVertex();
+			outerVertex->x(cos(angle) * halfWidth);
+			outerVertex->y(sin(angle) * halfHeight);
+			outerVertex->radius(cornerRadius());
+			addVertex(outerVertex);
+			angle += inc;
+
+			StraightVertex* innerVertex = new StraightVertex();
+			innerVertex->x(cos(angle) * innerHalfWidth);
+			innerVertex->y(sin(angle) * innerHalfHeight);
+			innerVertex->radius(cornerRadius());
+			addVertex(innerVertex);
+			angle += inc;
+		}
+	}
+
+	ParametricPath::update(value);
```

Comment:
Have to call this rather than `Super::update()` otherwise we get a polygon draw instead, since stars inherit from polygon. Not sure if there's a better way to handle this

Reply:
The core generator defines Base for you! Base::update should be the same.

Reply:
Ohh interesting, this is ok for now as it's a single case using this but if we start using this pattern more, we should probably define a virtual buildPolygon method that you can override.

Reply:
buildPolygon sounds like a nice way to do this, I'll refactor it as that

Reply:
refactored this into buildPolygon and buildVertex methods